x-pack/plugins/cases/public/components/all_cases/multi_select_filter.tsx

```diff
@@ -0,0 +1,161 @@
+  options.map((option) => option.label);
+
+const getEuiSelectableCheckedOptions = (options: EuiSelectableOption[]) =>
+  options.filter((option) => option.checked === 'on');
+
+interface UseFilterParams<T> {
+  buttonLabel?: string;
+  id: keyof FilterOptions;
+  limit?: number;
+  limitReachedMessage?: string;
+  onChange: ({ filterId, options }: { filterId: keyof FilterOptions; options: string[] }) => void;
+  options: string[];
+  selectedOptions?: string[];
+  renderOption?: (option: T) => React.ReactNode;
+}
+export const MultiSelectFilter = <T extends EuiSelectableOption>({
+  buttonLabel,
+  id,
+  limit,
+  limitReachedMessage,
+  onChange,
+  options: rawOptions,
+  selectedOptions = [],
+  renderOption,
+}: UseFilterParams<T>) => {
+  const { euiTheme } = useEuiTheme();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const toggleIsPopoverOpen = () => setIsPopoverOpen((prevValue) => !prevValue);
+  const isInvalid = Boolean(limit && limitReachedMessage && selectedOptions.length >= limit);
+  const options = fromRawOptionsToEuiSelectableOptions(rawOptions, selectedOptions);
+
+  useEffect(() => {
+    const trimmedSelectedOptions = selectedOptions.filter((option) => rawOptions.includes(option));
+    if (!isEqual(trimmedSelectedOptions, selectedOptions)) {
+      onChange({
+        filterId: id,
+        options: trimmedSelectedOptions,
+      });
+    }
+  }, [selectedOptions, rawOptions, id, onChange]);
+
+  const _onChange = (newOptions: EuiSelectableOption[]) => {
+    const newSelectedOptions = getEuiSelectableCheckedOptions(newOptions);
+    if (isInvalid && limit && newSelectedOptions.length >= limit) {
+      return;
+    }
+
+    onChange({
+      filterId: id,
+      options: fromEuiSelectableOptionToRawOption(newSelectedOptions),
+    });
+  };
+
+  return (
+    <EuiPopover
+      ownFocus
+      button={
+        <EuiFilterButton
+          data-test-subj={`options-filter-popover-button-${id}`}
+          iconType="arrowDown"
+          onClick={toggleIsPopoverOpen}
+          isSelected={isPopoverOpen}
+          numFilters={options.length}
+          hasActiveFilters={selectedOptions.length > 0}
+          numActiveFilters={selectedOptions.length}
+          aria-label={buttonLabel}
+        >
+          {buttonLabel}
+        </EuiFilterButton>
+      }
+      isOpen={isPopoverOpen}
+      closePopover={() => setIsPopoverOpen(false)}
+      panelPaddingSize="none"
+      repositionOnScroll
+    >
+      {isInvalid ? (
+        <>
+          <EuiHorizontalRule margin="none" />
+          <EuiCallOut
+            title={limitReachedMessage}
+            color="warning"
+            size="s"
+            data-test-subj="maximum-length-warning"
+          />
+          <EuiHorizontalRule margin="none" />
+        </>
+      ) : null}
+      <EuiSelectable<T>
+        options={options}
+        searchable
+        searchProps={{ placeholder: id, compressed: false }}
```

Comment:
Because we have `placeholder: id` here and `id` always starts in lowercase (`severity`, `status`, `category`, etc) the placeholder text will start in lowercase too. I am not sure this is correct design-wise 😅 

<img width="516" alt="Screenshot 2023-11-07 at 10 05 59" src="https://github.com/elastic/kibana/assets/1533137/6ecce658-8043-4c39-94ae-cb2d9d917fa7">

Reply:
Using the button label now